src/nimble.nim

```diff
@@ -1021,39 +1026,43 @@ proc developFromDir(dir: string, options: Options) =
-  if dirExists(pkgDestDir) and fileExists(pkgDestDir / "nimblemeta.json"):
-    let msg = "$1@$2 already exists. Overwrite?" %
-              [pkgInfo.name, pkgInfo.specialVersion]
-    if not options.prompt(msg):
-      raise NimbleQuit(msg: "")
-    removePkgDir(pkgDestDir, options)
-
-  createDir(pkgDestDir)
-  # The .nimble-link file contains the path to the real .nimble file,
-  # and a secondary path to the source directory of the package.
-  # The secondary path is necessary so that the package's .nimble file doesn't
-  # need to be read. This will mean that users will need to re-run
-  # `nimble develop` if they change their `srcDir` but I think it's a worthy
-  # compromise.
-  let nimbleLinkPath = pkgDestDir / pkgInfo.name.addFileExt("nimble-link")
-  let nimbleLink = NimbleLink(
-    nimbleFilePath: pkgInfo.myPath,
-    packageDir: pkgInfo.getRealDir()
-  )
-  writeNimbleLink(nimbleLinkPath, nimbleLink)
+  # Don't link if local deps mode and "developing" the top level package
+  if not (options.localdeps and options.startDir == dir):
+    # This is similar to the code in `installFromDir`, except that we
+    # *consciously* not worry about the package's binaries.
+    let pkgDestDir = pkgInfo.getPkgDest(options)
+    if dirExists(pkgDestDir) and fileExists(pkgDestDir / "nimblemeta.json"):
+      let msg = "$1@$2 already exists. Overwrite?" %
+                [pkgInfo.name, pkgInfo.specialVersion]
+      if not options.prompt(msg):
+        raise NimbleQuit(msg: "")
+      removePkgDir(pkgDestDir, options)
+
+    createDir(pkgDestDir)
+    # The .nimble-link file contains the path to the real .nimble file,
+    # and a secondary path to the source directory of the package.
+    # The secondary path is necessary so that the package's .nimble file doesn't
+    # need to be read. This will mean that users will need to re-run
+    # `nimble develop` if they change their `srcDir` but I think it's a worthy
+    # compromise.
+    let nimbleLinkPath = pkgDestDir / pkgInfo.name.addFileExt("nimble-link")
+    let nimbleLink = NimbleLink(
+      nimbleFilePath: pkgInfo.myPath,
+      packageDir: pkgInfo.getRealDir()
+    )
+    writeNimbleLink(nimbleLinkPath, nimbleLink)
 
-  # Save a nimblemeta.json file.
-  saveNimbleMeta(pkgDestDir, dir, vcsRevisionInDir(dir), nimbleLinkPath)
+    # Save a nimblemeta.json file.
+    saveNimbleMeta(pkgDestDir, dir, vcsRevisionInDir(dir), nimbleLinkPath)
 
-  # Save the nimble data (which might now contain reverse deps added in
-  # processDeps).
-  saveNimbleData(options)
+    # Save the nimble data (which might now contain reverse deps added in
+    # processDeps).
+    saveNimbleData(options)
 
-  display("Success:", (pkgInfo.name & " linked successfully to '$1'.") %
-          dir, Success, HighPriority)
+    display("Success:", (pkgInfo.name & " linked successfully to '$1'.") %
+            dir, Success, HighPriority)
+  else:
+    display("Warning:", "Skipping link in local deps mode", Warning)
```

Comment:
This should be an error, no?

Reply:
Similar to install, we are skipping creating a link of the top level project into its own deps directory.